[regression] 960f538819 change for the vp8 cause the regression on TGL

Catch the regression of decode/vp8 test case failed on daily test
with commit trace: 
```
commit 960f538819454487b43e20a1337e631b61c95b2d
Author: Wang, Yuchen <yuchen.wang@intel.com>
Date:   Tue Jul 6 11:14:09 2021 +0800

VPL Cleanup Opaque mem
```
Modify the mfx_vp8_dec_decode_hw.cpp And mfx_vp8_dec_decode_hw.h

the last success commit is 1aaad0ba6904

```
ffmpeg -init_hw_device qsv=qsv:hw -qsv_device /dev/dri/renderD128 -hwaccel qsv -filter_hw_device qsv -hwaccel_output_format qsv -v verbose -c:v vp8_qsv -i vp8.webm -vf 'hwdownload,format=nv12' -pix_fmt yuv420p -f rawvideo -vsync passthrough -autoscale 0 -vframes 2 -y vp8_64x64_I420.yuv
ffmpeg version N-102895-g2d84bcc2d8 Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.5.0-3ubuntu1~18.04)
  configuration: --prefix=/opt/media/install --disable-static --enable-shared --enable-libdrm --enable-vaapi --enable-libmfx --disable-amf --disable-audiotoolbox --disable-cuda --disable-cuda-sdk --disable-cuvid --disable-d3d11va --disable-dxva2 --disable-libnpp --disable-mmal --disable-nvdec --disable-nvenc --disable-omx --disable-omx-rpi --disable-rkmpp --disable-v4l2-m2m --disable-vdpau --disable-videotoolbox --enable-gpl --enable-libx264 --enable-libx265 --enable-libvpl
  libavcodec     59.  3.100 / 59.  3.100
  libavformat    59.  3.101 / 59.  3.101
  libavdevice    59.  0.100 / 59.  0.100
  libavfilter     8.  0.103 /  8.  0.103
  libswscale      6.  0.100 /  6.  0.100
  libswresample   4.  0.100 /  4.  0.100
  libpostproc    56.  0.100 / 56.  0.100
[AVHWDeviceContext @ 0x561c1b715a00] Trying to use DRM render node for device 0, with matching kernel driver (i915).
[AVHWDeviceContext @ 0x561c1b715a00] libva: VA-API version 1.13.0
[AVHWDeviceContext @ 0x561c1b715a00] libva: User requested driver 'iHD'
[AVHWDeviceContext @ 0x561c1b715a00] libva: Trying to open /opt/media/install/lib/dri/iHD_drv_video.so
[AVHWDeviceContext @ 0x561c1b715a00] libva: Found init function __vaDriverInit_1_13
[AVHWDeviceContext @ 0x561c1b715a00] libva: va_openDriver() returns 0
[AVHWDeviceContext @ 0x561c1b715a00] Initialised VAAPI connection: version 1.13
[AVHWDeviceContext @ 0x561c1b715a00] VAAPI driver: Intel iHD driver for Intel(R) Gen Graphics - 21.2.3 ().
[AVHWDeviceContext @ 0x561c1b715a00] Driver not found in known nonstandard list, using standard behaviour.
[AVHWDeviceContext @ 0x561c1b715480] Use the Intel oneVPL SDK to create MFX session, API version is 2.4, the required implementation version is 1.3
libva info: VA-API version 1.13.0
libva info: User environment variable requested driver 'iHD'
libva info: Trying to open /opt/media/install/lib/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_13
libva info: va_openDriver() returns 0
[AVHWDeviceContext @ 0x561c1b715480] Initialize MFX session: implementation version is 2.3
Input #0, matroska,webm, from '/opt/media/src/assets/otc-media/sanity/vp8/vp8_rev025_CoeffSkip_11_Citycorr.webm':
  Metadata:
    encoder         : vpxenc v0.9.6
  Duration: 00:00:00.07, start: 0.000000, bitrate: 463 kb/s
  Stream #0:0(eng): Video: vp8, 1 reference frame, yuv420p(progressive), 64x64, SAR 1:1 DAR 1:1, 30 fps, 30 tbr, 1k tbn (default)
Stream mapping:
  Stream #0:0 -> #0:0 (vp8 (vp8_qsv) -> rawvideo (native))
Press [q] to stop, [?] for help
[AVHWDeviceContext @ 0x7ff21c12c6c0] libva: VA-API version 1.13.0
[AVHWDeviceContext @ 0x7ff21c12c6c0] libva: User requested driver 'iHD'
[AVHWDeviceContext @ 0x7ff21c12c6c0] libva: Trying to open /opt/media/install/lib/dri/iHD_drv_video.so
[AVHWDeviceContext @ 0x7ff21c12c6c0] libva: Found init function __vaDriverInit_1_13
[AVHWDeviceContext @ 0x7ff21c12c6c0] libva: va_openDriver() returns 0
[AVHWDeviceContext @ 0x7ff21c12c6c0] Initialised VAAPI connection: version 1.13
[AVHWDeviceContext @ 0x7ff21c12c6c0] VAAPI driver: Intel iHD driver for Intel(R) Gen Graphics - 21.2.3 ().
[AVHWDeviceContext @ 0x7ff21c12c6c0] Driver not found in known nonstandard list, using standard behaviour.
[AVHWDeviceContext @ 0x7ff21c12c240] Use the Intel oneVPL SDK to create MFX session, API version is 2.4, the required implementation version is 1.3
libva info: VA-API version 1.13.0
libva info: User environment variable requested driver 'iHD'
libva info: Trying to open /opt/media/install/lib/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_13
libva info: va_openDriver() returns 0
[AVHWDeviceContext @ 0x7ff21c12c240] Initialize MFX session: implementation version is 2.3
[AVHWDeviceContext @ 0x7ff21c212180] VAAPI driver: Intel iHD driver for Intel(R) Gen Graphics - 21.2.3 ().
[AVHWDeviceContext @ 0x7ff21c212180] Driver not found in known nonstandard list, using standard behaviour.
[vp8_qsv @ 0x7ff21c000980] Decoder: output is video memory surface
[vp8_qsv @ 0x7ff21c000980] Use the Intel oneVPL SDK to create MFX session with the specified MFX loader
[AVHWDeviceContext @ 0x7ff21c212500] VAAPI driver: Intel iHD driver for Intel(R) Gen Graphics - 21.2.3 ().
[AVHWDeviceContext @ 0x7ff21c212500] Driver not found in known nonstandard list, using standard behaviour.
[vp8_qsv @ 0x7ff21c000980] Decoder: output is video memory surface
[vp8_qsv @ 0x7ff21c000980] Use the Intel oneVPL SDK to create MFX session with the specified MFX loader
<Error: AssertionError: CALL ERROR: failed with exitcode -11
```

--- a/_studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw.cpp
+++ b/_studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw.cpp
@@ -452,12 +452,13 @@ mfxStatus VideoDECODEVP8_HW::DecodeFrameCheck(mfxBitstream * /*p_bs*/, mfxFrameS
 mfxStatus VideoDECODEVP8_HW::GetOutputSurface(mfxFrameSurface1 **pp_surface_out, mfxFrameSurface1 *p_surface_work, UMC::FrameMemID index)
 {
     mfxFrameSurface1 *p_native_surface = m_surface_source->GetSurface(index, p_surface_work, &m_video_params);
-    (void)pp_surface_out;
 
     if (!p_native_surface)
     {
         return MFX_ERR_UNDEFINED_BEHAVIOR;
     }
+
+    *pp_surface_out = p_native_surface;
 
     return MFX_ERR_NONE;
 }

